plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/condition/ExactNumberOfProblemsExists.java

```diff
@@ -0,0 +1,75 @@
+package org.jboss.reddeer.eclipse.condition;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.reddeer.common.condition.WaitCondition;
+import org.jboss.reddeer.eclipse.ui.problems.Problem;
+import org.jboss.reddeer.eclipse.ui.problems.ProblemsView;
+import org.jboss.reddeer.eclipse.ui.problems.ProblemsView.ProblemType;
+import org.jboss.reddeer.eclipse.ui.problems.matcher.AbstractProblemMatcher;
+
+/**
+ * Wait condition expects existence of specific amount of problems in Problems view.
+ * 
+ * @author mlabuda@redhat.com
+ */
+public class ExactNumberOfProblemsExists implements WaitCondition {
+	
+	private AbstractProblemMatcher[] matchers;
+	private ProblemType problemType;
+	private int expectedProblemsCount;
+	
+	private List<Problem> problems = new ArrayList<Problem>();
+	
+	/**
+	 * Constructs the condition for the specified problem <var>type</var> and
+	 * the specified <var>count</var> of the problems.
+	 * 
+	 * @param type type of the problems
+	 * @param count number of the problems
+	 */
+	public ExactNumberOfProblemsExists(ProblemType type, int count) {
+		if(type == null) {
+			throw new NullPointerException("Problem Type cannot be null");
+		}
+		if(count < 0) {
+			throw new IllegalArgumentException("Number of problems must be greater than zero.");
+		}
+		
+		this.problemType = type;
+		this.expectedProblemsCount = count;
+		matchers = new AbstractProblemMatcher[]{};
+
```

Comment:
This is not needed because Problems view is opened in test() method. Consider to add field pointing to Problems view instantiate it in constructor and in test() method just call open() method on this instance